### DISCUSSION TITLE
channel From Path returns null even though file exists

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#3825] [Q&A]
Hi,
I wanted to create a channel with certain file as below:
```
nextflow.enable.dsl=1
channel.FromPath('/cluster/projects/p33/users/gledak/metabolomics/lipids/out/*_permuted*.glm.linear').view()
```
But it gives me error:
`Cannot invoke method view() on null object`

Any help? Attached is the log.
[nextflow.log3.txt](https://github.com/nextflow-io/nextflow/files/11129910/nextflow.log3.txt)

**Comment by manuelesimi** [comment]
You get a null object because you are invoking a method that does not exist.

It should be:
``
channel.fromPath('...').view()
``
Do note the lowercase f in fromPath.

**Comment by mribeirodantas** [comment]
The first letter in `Channel` must be uppercase, and the first letter in the channel factory `fromPath` must be lowercase 😃

Check below for the right syntax (The indentation, spacing, is optional but makes it easier to read so I'm going to use it here):

```Groovy
Channel
  .fromPath('/cluster/projects/p33/users/gledak/metabolomics/lipids/out/*_permuted*.glm.linear')
  .view()
```

It's also important to mention that DSL1 has been deprecated and is no longer supported in recent versions of Nextflow. You should use [DSL2](https://www.nextflow.io/docs/latest/dsl2.html) instead.

**Reply by manuelesimi** [reply]
`channel` is an alias of `Channel` since version 20.07.0 (see https://www.nextflow.io/docs/latest/channel.html#channel-factory), so it should work.

Agree that DSL1 should be avoided as much as possible.

**Reply by mribeirodantas** [reply]
Thanks for pointing this out, @manuelesimi. I completely forgot about it 😓

**Comment by nick-youngblut** [comment]
I've developed many Nextflow pipelines, and for starting a new pipeline, I'm stuck on this simple aspect:

```
myFileChannel = Channel.fromPath( '/data/some/bigfile.txt', checkIfExists = true )
println( myFileChannel )
```

...returns `null`, and there is no check on whether the file exists.

I'm using Nextflow v24.04.4.  What is going on???

**Reply by nick-youngblut** [reply]
It all came down to `checkIfExists = true` instead of `checkIfExists: true`. The misleading (or lacking) error messages in Nextflow can waste so much time.

**Reply by nick-youngblut** [reply]
Maybe this a groovy-level issue, but `Channel.fromPath( '/data/some/bigfile.txt', checkIfExists = true )` should really throw an error instead of returning `null`.

**Reply by bentsherman** [reply]
It will soon. Stay tuned 😄
